### PR TITLE
doc: Algorithm README in en

### PR DESCRIPTION
**Pull request by wey-gu**
Algorithm Read Me in English.

Triggered via https://github.com/vesoft-inc/nebula-java/issues/299

### CODE REVIEW

**Comment by wey-gu** [comment]
Sorry, I would like to make it quick, to avoid silly translations, I didn't translate the column of the use cases, could you @Nicole00  help do that on top of this PR?

**Comment by Nicole00** [comment]
Great Job, thank you~
@randomJoe211 Joe, please help to review this readme.

File: nebula-algorithm/README.md [hunk 1]
```diff
@@ -1,75 +1,83 @@
...
+
+<p align="center">
+  <br>English | <a href="README-CN.md">中文</a>
+</p>
+
+nebula-algorithm is a Spark Application based on [GraphX](https://spark.apache.org/graphx/) with below Algorithm provided for now:
+
+
+|          Name          |Use Case|
+|:------------------------:|:---------------:|
```
Comment (Nicole00) [inline nebula-algorithm/README.md:11]: The use case for pagerank are page ranking, important node digging.
The use case for louvain are community digging, hierarchical clustering.
The use case for kcore are community detection, financial risk control.
The use case for labelpropagation are community detection, consultation propagation, advertising recommendation.
The use case for connectedcomponent are community detection, isolated island detection.
The use case for stronglyconnectedcomponent is community detection.
The use case for shortestpath are path plan, network plan.
The use case for trianglecount is network structure analysis.
The use case for betweennesscentrality are important node digging, node influence calculation.
The use case for degreestatic is graph structure analysis.

File: nebula-algorithm/README.md [hunk 2]
```diff
@@ -1,75 +1,83 @@
...
+|         Louvain          | 社团挖掘、层次化聚类|
+|          KCore             |社区发现、金融风控|
+|     LabelPropagation     |资讯传播、广告推荐、社区发现|
+|    ConnectedComponent    |社区发现、孤岛发现|
+|StronglyConnectedComponent|社区发现|
+|       ShortestPath        |路径规划、网络规划|
+|       TriangleCount      |网络结构分析|
+|   BetweennessCentrality  |关键节点挖掘，节点影响力计算|
+
+You could either leverage comprehensive Nebula-Algorithm tooling in way to submit its `Spark Application` or invoke its `lib` in a programming way to apply Graph Compute towards the `DataFrame`.
```
Comment (Nicole00) [inline nebula-algorithm/README.md:22]: You could submit the entire spark application or invoke algorithms in `lib` library to apply graph algorithms for DataFrame.

Comment (wey-gu) [reply nebula-algorithm/README.md]: Added.

File: nebula-algorithm/README.md (hunk 1 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:11]: Added.

File: nebula-algorithm/README.md [hunk 3]
```diff
@@ -1,75 +1,85 @@
...
- |       ShortestPath       |  最短路径   |路径规划、网络规划|
- |       TriangleCount      | 三角形计数  |网络结构分析|
- |   BetweennessCentrality  | 介数中心性  |关键节点挖掘，节点影响力计算|
- |        DegreeStatic      |   度统计   |图结构分析|
- 
-使用 `nebula-algorithm`，可以通过提交 `Spark` 任务的形式使用完整的算法工具对 `Nebula Graph` 数据库中的数据执行图计算，也可以通过编程形式调用`lib`库下的算法针对DataFrame执行图计算。
-
-## 如何获取
- 1. 编译打包 Nebula-Algorithm
+# Welcome to Nebula-Algorithm
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:1]: Nebula Algorithm

File: nebula-algorithm/README-CN.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# 欢迎使用 nebula-algorithm
```
Comment (randomJoe211) [inline nebula-algorithm/README-CN.md:1]: Nebula Algorithm. Please modify the product names in the following content as well.

File: nebula-algorithm/README.md [hunk 5]
```diff
@@ -1,75 +1,85 @@
...
+
+<p align="center">
+  <br>English | <a href="README-CN.md">中文</a>
+</p>
+
+nebula-algorithm is a Spark Application based on [GraphX](https://spark.apache.org/graphx/) with below Algorithm provided for now:
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:7]: Lacking article: the below Algorithm.
And in Nebula docs we use "following" more often.

File: nebula-algorithm/README.md [hunk 6]
```diff
@@ -1,75 +1,85 @@
...
+
+## Get Nebula-Algorithm
+ 1. Build Nebula-Algorithm
     ```
     $ git clone https://github.com/vesoft-inc/nebula-spark-utils.git
     $ cd nebula-algorithm
     $ mvn clean package -Dgpg.skip -Dmaven.javadoc.skip=true -Dmaven.test.skip=true
     ```
-    编译完成后，在 `nebula-algorithm/target` 目录下会生成 `nebula-algorithm-2.0.0.jar` 。
+    After the above buiding process the target file  `nebula-algorithm-2.0.0.jar` will be placed under `nebula-algorithm/target` .
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:33]: After the above building process,

Comment (randomJoe211) [reply nebula-algorithm/README.md:33]: `nebula-algorithm/target`.
There's one extra space.

File: nebula-algorithm/README.md [hunk 7]
```diff
@@ -1,75 +1,85 @@
...
 
- 2. 在 Maven 远程仓库下载
-   https://repo1.maven.org/maven2/com/vesoft/nebula-algorithm/2.0.0/
+ 2. Download from Maven repo
+      
+      Alternatively, it could be downlaoded from below Maven repo:
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:37]: downloaded (typo) from the below/following Maven repo

File: nebula-algorithm/README.md [hunk 8]
```diff
@@ -1,75 +1,85 @@
...
-* 使用方法1：直接提交 nebula-algorithm 算法包
+Limitation: Due to Nebula-Algorithm will not encode string id, thus during the algorithm execution, the source and target of edges must be in Type Int (The `vid_type` in Nebula Space could be String, while data must be in Type Int).
+
+* Option 1: Submit nebula-algorithm package
 
-   * 设置配置文件
+   * Configuration
    
-    关于配置项的具体说明参考[示例配置](https://github.com/vesoft-inc/nebula-spark-utils/blob/master/nebula-algorithm/src/main/resources/application.conf)
+   Refer to [configuration example](https://github.com/vesoft-inc/nebula-spark-utils/blob/master/nebula-algorithm/src/main/resources/application.conf)
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:49]: Refer to the [configuration example]().

File: nebula-algorithm/README.md [hunk 9]
```diff
@@ -1,75 +1,85 @@
...
 
-   * 提交算法任务
+   * Submit Spark Applicaiton
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:51]: typo: Application

File: nebula-algorithm/README.md [hunk 10]
```diff
@@ -1,75 +1,85 @@
...
+   
+    For other algorithms, please refer to [test cases](https://github.com/vesoft-inc/nebula-spark-utils/tree/master/nebula-algorithm/src/test/scala/com/vesoft/nebula/algorithm/lib).
+   
+   *Note: The first column of DataFrame in the application are the source vertices, the second are the target vertices and the third are edges' weight.*
 
-## 贡献
+## Contribute
 
-nebula-algorithm 是一个完全开源的项目，欢迎开源爱好者通过以下方式参与：
+nebula-algorithm is an Open-Source project, you are more than welcomed to contribute in below ways:
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:81]: in the below/following ways

File: nebula-algorithm/README.md [hunk 11]
```diff
@@ -1,75 +1,85 @@
...
     ```
-* 使用方法2：调用 nebula-algorithm 算法接口
+   
+* Option2: Call nebula-algorithm interface
 
-   在`nebula-algorithm`的`lib`库中提供了10中常用图计算算法，可通过编程调用的形式调用算法。
-   * 在pom.xml中添加依赖
+   Now there are 10 algorithms provided in `lib` from `nebula-algorithm`, which could be invoked in a programming fashion as below:
+   
+   * Add depedencies from `pom.xml`
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:61]: dependencies

File: nebula-algorithm/README.md [hunk 12]
```diff
@@ -1,75 +1,85 @@
...
    val louvainResult = PageRankAlgo.apply(spark, data, prConfig, false)
    ```
- 
-    其他算法的调用方法见[测试示例](https://github.com/vesoft-inc/nebula-spark-utils/tree/master/nebula-algorithm/src/test/scala/com/vesoft/nebula/algorithm/lib) 。
-    
-    *注：执行算法的DataFrame默认第一列是源点，第二列是目标点，第三列是边权重。*
+   
+    For other algorithms, please refer to [test cases](https://github.com/vesoft-inc/nebula-spark-utils/tree/master/nebula-algorithm/src/test/scala/com/vesoft/nebula/algorithm/lib).
+   
+   *Note: The first column of DataFrame in the application are the source vertices, the second are the target vertices and the third are edges' weight.*
```
Comment (randomJoe211) [inline nebula-algorithm/README.md:77]: Plural form problem:
The first column of DataFrame in the application represents the source vertices
or
The first column of DataFrame in the application lists the source vertices

So are "the second are xxx" and " the third are".

Comment (randomJoe211) [reply nebula-algorithm/README.md:77]: Use the markdown quote for notes would be a better idea.

> **Note:**

File: nebula-algorithm/README.md (hunk 10 above)
Comment (randomJoe211) [reply nebula-algorithm/README.md:81]: "open source" is good enough.

**Comment by randomJoe211** [comment]
👍

File: nebula-algorithm/README.md (hunk 3 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:1]: Will do for all.

File: nebula-algorithm/README-CN.md (hunk 4 above)
Comment (wey-gu) [reply nebula-algorithm/README-CN.md:1]: sure.

File: nebula-algorithm/README.md (hunk 8 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:49]: done

File: nebula-algorithm/README.md (hunk 7 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:37]: done

File: nebula-algorithm/README.md (hunk 6 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:33]: both done

File: nebula-algorithm/README.md (hunk 10 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:81]: done

File: nebula-algorithm/README.md (hunk 9 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:51]: done, sorry

File: nebula-algorithm/README.md (hunk 12 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:77]: Done, thanks!

**Comment by Nicole00** [comment]
👍

File: nebula-algorithm/README.md (hunk 11 above)
Comment (wey-gu) [reply nebula-algorithm/README.md]: done,sorry

File: nebula-algorithm/README.md [hunk 13]
```diff
@@ -1,75 +1,85 @@
...
     ```
-* 使用方法2：调用 nebula-algorithm 算法接口
+   
+* Option2: Call nebula-algorithm interface
 
-   在`nebula-algorithm`的`lib`库中提供了10中常用图计算算法，可通过编程调用的形式调用算法。
-   * 在pom.xml中添加依赖
+   Now there are 10 algorithms provided in `lib` from `nebula-algorithm`, which could be invoked in a programming fashion as below:
+   
+   * Add dependencies from `pom.xml`
```
Comment (Nicole00) [inline nebula-algorithm/README.md:61]: Add dependencies in `pom.xml`

File: nebula-algorithm/README.md [hunk 14]
```diff
@@ -1,75 +1,85 @@
...
    ```
     <dependency>
          <groupId>com.vesoft</groupId>
          <artifactId>nebula-algorithm</artifactId>
          <version>2.0.0</version>
     </dependency>
    ```
-   * 定义算法参数调用算法（以`PageRank`为例）
+   * Pass arguments of the algorithm, below is an example for `PageRank`.
```
Comment (Nicole00) [inline nebula-algorithm/README.md:69]: Instantiate algorithm‘s config,

File: nebula-algorithm/README.md (hunk 13 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:61]: done

File: nebula-algorithm/README.md (hunk 14 above)
Comment (wey-gu) [reply nebula-algorithm/README.md:69]: done

**Review by Nicole00 — APPROVED** [review]

**Pull request merged**